lib/ansible/module_utils/basic.py

```diff
@@ -569,6 +590,182 @@ def tmpdir(self):
 
         return self._tmpdir
 
+    # Begin Backwards compatibility only methods
+    # The follow methods should not be used directly. Use the public function
+    # or ArgumentSpecValidator class instead.
+    def _get_wanted_type(self, wanted, k):
+        # Use the private method for 'str' type to handle the string conversion warning.
```

Comment:
Can we deprecate these? (Or is this planned for a follow-up PR?)

Reply:
The longer term strategy should be to deprecate `basic.py` in favor of a new interface. Most of the code in here has been around so long that people are using/abusing it in interesting ways. But since these private methods already were removed, seems like we're going to just stick with that change.